client/attest.go

```diff
@@ -183,6 +191,49 @@ func (d *SevSnpDevice) Close() error {
 	return nil
 }
 
+// CreateTdxDevice opens the TDX attestation driver and wraps it with behavior
+// that allows it to add an attestation quote to pb.Attestation.
+func CreateTdxDevice() (*TdxDevice, error) {
+	d, err := tg.OpenDevice()
+	if err != nil {
+		return nil, err
+	}
+	return &TdxDevice{Device: d}, nil
+}
+
+// AddAttestation will get the TDX attestation quote given opts.TEENonce
+// and add them to `attestation`. If opts.TEENonce is empty, then uses
+// contents of opts.Nonce.
+func (d *TdxDevice) AddAttestation(attestation *pb.Attestation, opts AttestOpts) error {
+	var tdxNonce [tabi.TdReportDataSize]byte
+	if len(opts.TEENonce) == 0 {
+		copy(tdxNonce[:], opts.Nonce[:])
+	} else if len(opts.TEENonce) != tabi.TdReportDataSize {
+		return fmt.Errorf("the TEENonce size is %d. Intel TDX device requires 64", len(opts.TEENonce))
```

Comment:
Nit: don't hardcode 64, use tabi.TdReportDataSize in the message

Reply:
Done